azkaban-common/src/main/java/azkaban/imagemgmt/version/VersionInfo.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.imagemgmt.version;
+
+import azkaban.imagemgmt.models.ImageVersion.State;
+import com.google.common.base.Preconditions;
+import java.util.Objects;
+import org.codehaus.jackson.annotate.JsonCreator;
+import org.codehaus.jackson.annotate.JsonProperty;
+
+/**
+ * This class represents image version specific info such as version, path and current state.
+ * This POJO class is stored as a version information for the given image type in VersionSet.
+ */
+public class VersionInfo {
+
+  private final String version;
+  private final String path;
```

Comment:
Can the path be null or empty ?
Could you please add a comment describing these assumptions, especially what's the expected value for 'images' which don't need a path, such as az_config.

Reply:
Yes, I have comment for the path and it can have null or empty value for azkaban-config.